include/bluetooth/mesh/gen_battery_cli.h

```diff
@@ -83,6 +83,9 @@ struct bt_mesh_battery_cli {
  * function will return, and the response will be passed to the
  * @ref bt_mesh_battery_cli::status_handler callback.
  *
+ * @note Blocking calls must be made from the main cooperative thread only. Do not use in
+ * button handlers or workqueues.
```

Comment:
They probably _can_ be used in a workqueue, if you set it up to run in a thread with correct priority, I think. Suggest to change to
```suggestion
 * button handlers or the system workqueue.
```